From 3b1874d07f80634a2756fcd41b1387859d981d62 Mon Sep 17 00:00:00 2001
From: Robby Cai <robby.cai@nxp.com>
Date: Wed, 29 Nov 2017 16:18:15 +0800
Subject: [PATCH 2952/5242] MLK-17025-2 dts: imx8mq-evk: use internal clock as
 MCLK source for camera

commit  72d7e500598f90ce42ab7c813a196379d1c7f1b4 from
https://source.codeaurora.org/external/imx/linux-imx.git

use internal CLKO2 as camera's MCLK. it's 20MHz, derived from the
parent clock IMX8MQ_SYS2_PLL_200M

Signed-off-by: Robby Cai <robby.cai@nxp.com>
Reviewed-by: Sandor Yu <sandor.yu@nxp.com>
(cherry picked from commit 6bf9f6c7d0bcc62bd526bfb112bab91e916274b6)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts |   17 ++++++++++++-----
 1 file changed, 12 insertions(+), 5 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
index 40641c0..9a52180 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
@@ -459,12 +459,16 @@
 		status = "okay";
 		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_csi1>;
-		clocks = <&clk IMX8MQ_CLK_DUMMY>;
+		clocks = <&clk IMX8MQ_CLK_CLKO2_DIV>;
 		clock-names = "csi_mclk";
+		assigned-clocks = <&clk IMX8MQ_CLK_CLKO2_SRC>,
+				  <&clk IMX8MQ_CLK_CLKO2_DIV>;
+		assigned-clock-parents = <&clk IMX8MQ_SYS2_PLL_200M>;
+		assigned-clock-rates = <0>, <20000000>;
 		csi_id = <0>;
 		pwn-gpios = <&gpio1 3 GPIO_ACTIVE_HIGH>;
 		rst-gpios = <&gpio1 6 GPIO_ACTIVE_HIGH>;
-		mclk = <24000000>;
+		mclk = <20000000>;
 		mclk_source = <0>;
 		port {
 			ov5640_mipi1_ep: endpoint {
@@ -479,13 +483,16 @@
 		status = "disabled";
 		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_csi2>;
-		clocks = <&clk IMX8MQ_CLK_DUMMY>;
+		clocks = <&clk IMX8MQ_CLK_CLKO2_DIV>;
 		clock-names = "csi_mclk";
+		assigned-clocks = <&clk IMX8MQ_CLK_CLKO2_SRC>,
+				  <&clk IMX8MQ_CLK_CLKO2_DIV>;
+		assigned-clock-parents = <&clk IMX8MQ_SYS2_PLL_200M>;
+		assigned-clock-rates = <0>, <20000000>;
 		csi_id = <0>;
 		pwn-gpios = <&gpio1 5 GPIO_ACTIVE_HIGH>;
 		rst-gpios = <&gpio1 6 GPIO_ACTIVE_HIGH>;
-		/*AVDD-supply = <&vgen6_reg>;*/
-		mclk = <24000000>;
+		mclk = <20000000>;
 		mclk_source = <0>;
 		port {
 			ov5640_mipi2_ep: endpoint {
-- 
1.7.9.5

